src/main.c

```diff
@@ -2078,12 +2096,17 @@ static gboolean service_start(int argc, char **argv)
 			g_message("Restarted RAUC service");
 			r_event_log_message(R_EVENT_LOG_TYPE_SERVICE, "Service restarted");
 		} else {
-			RaucSlot *booted_slot = r_slot_find_by_device(r_context()->config->slots, r_context()->bootslot);
-			if (!booted_slot)
-				booted_slot = r_slot_find_by_bootname(r_context()->config->slots, r_context()->bootslot);
-			r_slot_status_load(booted_slot);
+			if (g_strcmp0(r_context()->bootslot, "_external_") == 0) {
+				r_event_log_booted_external();
+			} else {
+				RaucSlot *booted_slot = r_slot_find_by_device(r_context()->config->slots, r_context()->bootslot);
+				if (!booted_slot)
+					booted_slot = r_slot_find_by_bootname(r_context()->config->slots, r_context()->bootslot);
 
-			r_event_log_booted(booted_slot);
+				r_slot_status_load(booted_slot);
+
+				r_event_log_booted(booted_slot);
+			}
```

Comment:
Shouldn't we try to handle the case where we still can't find the booted slot?

We may not be able to install in that state, but `rauc info` should still work.

Reply:
I've thought about this, too but did not find a proper handling.

If the slot is invalid, the service will abort anyway with

    Failed to determine slot states: Did not find booted slot (matching 'XYZ')

In how far do you mean that `rauc info` will not work? It does not use the service, thus this should be unrelated.